### PR TITLE
feat: generate tmp password & save in redis

### PR DESCRIPTION
**Pull request by clothe09986**

### CODE REVIEW

**Comment by clothe09986** [comment]
[【帳戶】萬能密碼改版為申請制](https://trello.com/c/4yhzEW7H/4589-%E3%80%90%E5%B8%B3%E6%88%B6%E3%80%91%E8%90%AC%E8%83%BD%E5%AF%86%E7%A2%BC%E6%94%B9%E7%89%88%E7%82%BA%E7%94%B3%E8%AB%8B%E5%88%B6)

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -19,4 +24,33 @@ export class AuthController {
...
+      const { data: memberData } = await this.memberService.getMembersByCondition(appId, { limit: 1 }, { email });
+      if (memberData.length === 0) {
+        return {
+          code: 'E_NO_MEMBER',
+          message: 'member not found',
+          result: null,
+        };
+      }
+      const result = await this.authService.generateTmpPassword(appId, email);
+      await this.authService.insertAuthAuditLog(applicant, memberData[0].id, purpose);
```
Comment (jackey8616) [inline src/auth/auth.controller.ts]: Controller should not contains business logics.

File: src/entity/PermissionGroup.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Column, Entity, Index, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
-import { MemberPermissionGroup } from './MemberPermissionGroup';
+import { MemberPermissionGroup } from '../member/entity/member_permission_group.entity';
```
Comment (jackey8616) [inline src/entity/PermissionGroup.ts]: `'~/member/entity/member_permission_group.entity'`

File: src/entity/index.ts [hunk 3]
```diff
@@ -84,7 +84,7 @@ import { MemberDevice } from './MemberDevice';
 import { MemberNote } from './MemberNote';
 import { MemberOauth } from './MemberOauth';
 import { MemberPermissionExtra } from './MemberPermissionExtra';
-import { MemberPermissionGroup } from './MemberPermissionGroup';
+import { MemberPermissionGroup } from '../member/entity/member_permission_group.entity';
```
Comment (jackey8616) [inline src/entity/index.ts]: '~/member/entity/member_permission_group.entity'

File: src/member/entity/member_permission_group.entity.ts [hunk 4]
```diff
@@ -1,13 +1,19 @@
 import { Column, Entity, Index, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
 import { Member } from '~/member/entity/member.entity';
-import { PermissionGroup } from './PermissionGroup';
+import { PermissionGroup } from '../../entity/PermissionGroup';
```
Comment (jackey8616) [inline src/member/entity/member_permission_group.entity.ts]: `~/entity/PermissionGroup'`

File: test/auth.e2e-spec.ts [hunk 5]
```diff
@@ -2,85 +2,141 @@ import { INestApplication } from '@nestjs/common';
...
-    //     })
-    //     .expect(200);
-      
-    //   const { authToken } = body;
-    //   expect(body).toStrictEqual({
-    //     code: 'SUCCESS',
-    //     message: 'get auth token successfully',
-    //     result: { authToken },
-    //   });
-    // });
```
Comment (jackey8616) [inline test/auth.e2e-spec.ts]: I want my comment back.

File: test/auth.e2e-spec.ts [hunk 6]
```diff
@@ -2,85 +2,141 @@ import { INestApplication } from '@nestjs/common';
...
+        applicant: member.id,
+        email: member.email,
+        purpose: 'test',
+      });
+
+      expect(body.code).toBe('SUCCESS');
+      expect(body.result.password).toBeDefined();
+    });
+
+    it('should fail to retrieve a temporary password', async () => {
```
Comment (jackey8616) [inline test/auth.e2e-spec.ts]: 'should fail to retrieve a temporary password'
-> Describe with situation
e.g.: 'Should fail due to ......'

File: test/auth.e2e-spec.ts [hunk 7]
```diff
@@ -2,85 +2,141 @@ import { INestApplication } from '@nestjs/common';
...
+        purpose: 'test',
+      });
+      expect(body).toStrictEqual({
+        code: 'E_TMP_PASSWORD',
+        message: 'failed to generate temporary password',
+        result: null,
+      });
+    });
+
+    it('should insert an audit log', async () => {
```
Comment (jackey8616) [inline test/auth.e2e-spec.ts]: Combine with success cases.

File: test/auth.e2e-spec.ts [hunk 8]
```diff
@@ -2,85 +2,141 @@ import { INestApplication } from '@nestjs/common';
...
+        email: member.email,
+        purpose: 'test',
+      });
+
+      const authAuditLog = await manager.getRepository(AuthAuditLog).findOne({ where: { memberId: member.id } });
+
+      expect(authAuditLog.action).toBe('apply_temporary_password');
+    });
+
+    it('should insert the temporary password into Redis', async () => {
```
Comment (jackey8616) [inline test/auth.e2e-spec.ts]: Combine with success cases.

File: src/auth/auth.type.ts [hunk 9]
```diff
@@ -3,3 +3,10 @@ export interface CrossServerTokenDTO {
   key: string;
   permissions: Array<string>;
 }
+
+export interface GenerateTmpPasswordDto {
```
Comment (jackey8616) [inline src/auth/auth.type.ts]: DTO.

**Review by jackey8616 — CHANGES_REQUESTED** [review]

File: src/auth/auth.controller.ts (hunk 1 above)
Comment (clothe09986) [reply src/auth/auth.controller.ts]: fixed

**Review by jackey8616 — APPROVED** [review]

**Pull request merged**